### PR TITLE
Paddlepaddle init

### PR DESCRIPTION
**Pull request by happysalada**
###### Description of changes

having an initial stab at paddlepaddle.
This is far from perfect.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 243583` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>python310Packages.paddle_bfloat</li>
    <li>python310Packages.paddle_bfloat.dist</li>
    <li>python310Packages.paddlepaddle</li>
    <li>python310Packages.paddlepaddle.dist</li>
    <li>python311Packages.paddle_bfloat</li>
    <li>python311Packages.paddle_bfloat.dist</li>
  </ul>
</details>

**Comment by happysalada** [comment]
@natsukium hey sorry to bother you, I'm kind of stuck on this package.
The error on ofborg points to the following issue on github https://github.com/PaddlePaddle/Paddle/issues/48681
The issues talks about paddle_gpu though, which doesn't make a lot of sense to me.
I was packaging this to get to paddleOCR eventually, apparently it's one of the best OCRs around.
if this looks too bad, feel free to say so, I'm happy to just close this. I just thought I might ask you first.

File: pkgs/development/python-modules/paddlepaddle/default.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    dist = pyShortVersion;
+    python = pyShortVersion;
+    abi = pyShortVersion;
+  } // hashAndPlatform);
+in
+buildPythonPackage {
+  inherit pname version format src;
+
+  disabled = pythonOlder "3.9" || pythonAtLeast "3.11";
+
```
Comment (natsukium) [inline pkgs/development/python-modules/paddlepaddle/default.nix:36]:
```suggestion

  nativeBuildInputs = [
    autoPatchelfHook
  ];
  
  buildInputs = [
    openssl_1_1
    zlib
  ];
  
```

We can use `autoPatchelfHook` here.
However, openssl_1_1 is already deprecated in nixpkgs, so we need to do something about it.

**Comment by natsukium** [comment]
Can we build from the source following this instruction?
I think the openssl issue may be solved.
https://www.paddlepaddle.org.cn/documentation/docs/en/install/compile/linux-compile_en.html#compile-on-linux-from-source-code

**Comment by Avaq** [comment]
FYI: I'm working off this branch to get PaddleOCR packaged. See [my commits](https://github.com/NixOS/nixpkgs/compare/master...Avaq:nixpkgs:avaq/paddleocr) for the progress. At the time of writing, I have PaddleOCR working as a *python package* through which I got some text extraction using:

```python
from paddleocr import PaddleOCR

ocr = PaddleOCR(use_angle_cls=True, lang='en', show_log=False)
ocr.ocr('./ocr-test-image.jpg', cls=True)
```

**Comment by natsukium** [comment]
great work, @Avaq!
I look forward to your PR, even a draft is fine.

**Pull request closed without merging**